src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioBaseDiagnosticListTable.LiveTableDataSource.cs

```diff
@@ -37,7 +37,9 @@ protected class LiveTableDataSource : AbstractRoslynTableDataSource<DiagnosticTa
             private readonly Workspace _workspace;
             private readonly OpenDocumentTracker<DiagnosticTableItem> _tracker;
 
-            public LiveTableDataSource(Workspace workspace, IDiagnosticService diagnosticService, string identifier)
+            private bool _isBuildRunning;
```

Comment:
threading docs on this sort of thing would be good.

Reply:
Sure, I'll add comments. Build progress updates, just like solution crawler progress updates, are all serialized by using task queue. They may occur on any thread though.